src/taskgraph/morph.py

```diff
@@ -52,16 +52,17 @@ def amend_taskgraph(taskgraph, label_to_taskid, to_add):
 def derive_index_task(task, taskgraph, label_to_taskid, parameters, graph_config):
     """Create the shell of a task that depends on `task` and on the given docker
     image."""
+    task_index_config = graph_config["taskgraph"].get("task-index", {})
+    eager = task_index_config.get("eager", False)
```

Comment:
We probably want this to be a per-task setting rather than global all or nothing setting.

Reply:
Note we have an `index_builder` mechanism in the `task.py` transforms already, e.g:
https://github.com/taskcluster/taskgraph/blob/main/src/taskgraph/transforms/task.py#L889

Maybe we can add this `eager` flag over there instead.